pkg/logs/agent.go

```diff
@@ -114,6 +118,39 @@ func (a *Agent) Stop() {
 		// trying to write to the network.
 		a.destinationsCtx.Stop()
 		// Wait again for the stopper to complete.
-		<-c
+		// In some situation, the stopper unfortunately never succeed to complete,
+		// we've already reached the grace period, give it some more seconds and
+		// then force quit.
+		timeout := time.NewTimer(5 * time.Second)
+		select {
+		case <-c:
+		case <-timeout.C:
+			log.Debug("Force close of the Logs Agent, dumping the stack trace.")
+			if stack, err := getAgentStackTrace(); err != nil {
+				log.Debugf("can't get the stack trace of the Agent: %s\n", err)
+			} else {
+				log.Debug(string(stack))
+			}
+		}
 	}
 }
+
+// getStackTrace returns the stack trace of a running Agent.
+// Used as a best effort to get info while force closing the Logs Agent.
+func getAgentStackTrace() ([]byte, error) {
+	pprofURL := fmt.Sprintf("http://127.0.0.1:%s/debug/pprof/goroutine?debug=2",
```

Comment:
Can you be affected by the issue solved by https://github.com/DataDog/datadog-agent/pull/4194?

Reply:
Nice catch, I think you're right. I'll look into that!